### PR TITLE
hal: renesas: ra: Add support for SDHC driver

### PR DESCRIPTION
**Pull request by quytranpzz**
Add support for SDHC driver with FSP r_sdhi

### CODE REVIEW

File: drivers/ra/CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ zephyr_library_sources_ifdef(CONFIG_USE_RA_FSP_FLASH_HP
 zephyr_library_sources_ifdef(CONFIG_USE_RA_FSP_SPI_B
 	fsp/src/r_spi_b/r_spi_b.c)
 zephyr_library_sources_ifdef(CONFIG_USE_RA_FSP_SPI
-	fsp/src/r_spi/r_spi.c)	
+	fsp/src/r_spi/r_spi.c)
```
Comment (KhiemNguyenT) [inline drivers/ra/CMakeLists.txt:39]: This change should be excluded.

File: drivers/ra/README [hunk 2]
```diff
@@ -58,3 +58,7 @@ Patch List:
    * Move the include for cmsis_compiler.h after include for Renesas CMSIS header file
    Impacted files:
       drivers/ra/fsp/src/bsp/cmsis/Device/RENESAS/Include/renesas.h
+
+   * Remove the static definition in SDHI
```
Comment (KhiemNguyenT) [inline drivers/ra/README:62]: This change should be separated in another patch.

**Review by KhiemNguyenT — CHANGES_REQUESTED** [review]

File: drivers/ra/CMakeLists.txt (hunk 1 above)
Comment (quytranpzz) [reply drivers/ra/CMakeLists.txt]: I removed the change. Thank you

File: drivers/ra/README (hunk 2 above)
Comment (quytranpzz) [reply drivers/ra/README:62]: I've split it into a new commit. Thanks.

**Review by KhiemNguyenT — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Comment by quytranpzz** [comment]
rebase with the main branch

**Review by KhiemNguyenT — APPROVED** [review]
